test/utils/update/fetch.managing.officers.private.data.spec.ts

```diff
@@ -0,0 +1,127 @@
+
+    expect(mockGetManagingOfficersPrivateData).not.toHaveBeenCalled();
+  });
+
+  test("should handle absence of private Managing Officer details", async () => {
+    mockIsActiveFeature.mockReturnValue(false);
+    mockGetManagingOfficersPrivateData.mockResolvedValue([]);
+
+    await fetchManagingOfficersPrivateData(appData, req);
+
+    expect(mockLoggerInfo).toHaveBeenCalledWith(expect.stringContaining("No private Managing Officer details were retrieved"));
+  });
+
+  test("should handle error when fetching Managing Officer private data fails", async () => {
+    mockIsActiveFeature.mockReturnValue(false);
+    mockGetManagingOfficersPrivateData.mockRejectedValue(new Error("some error"));
+
+    await fetchManagingOfficersPrivateData(appData, req);
+
+    expect(mockLoggerError).toHaveBeenCalledWith(req, expect.stringContaining("Private Managing Officer details could not be retrieved"));
+  });
+
+  test("should not fetch MO private data if email address is present", async () => {
+    appData = { entity: { email: "email@email.com" } };
+
+    await fetchManagingOfficersPrivateData(appData, req);
+    expect(mockGetManagingOfficersPrivateData).not.toHaveBeenCalled();
+    expect(appData.update?.review_managing_officers_individual?.length).toBeUndefined();
+    expect(appData.update?.review_managing_officers_corporate?.length).toBeUndefined();
+  });
+
+  test("should not fetch MO private data if overseasEntityId undefined", async () => {
+    appData = { overseas_entity_id: undefined };
+
+    await fetchManagingOfficersPrivateData(appData, req);
+    expect(mockGetManagingOfficersPrivateData).not.toHaveBeenCalled();
+    expect(appData.update?.review_managing_officers_individual?.length).toBeUndefined();
+    expect(appData.update?.review_managing_officers_corporate?.length).toBeUndefined();
+  });
+
+  test("should not fetch MO private data if transactionId undefined", async () => {
+    appData = { transaction_id: undefined };
+
+    await fetchManagingOfficersPrivateData(appData, req);
+    expect(mockGetManagingOfficersPrivateData).not.toHaveBeenCalled();
+    expect(appData.update?.review_managing_officers_individual?.length).toBeUndefined();
+    expect(appData.update?.review_managing_officers_corporate?.length).toBeUndefined();
+  });
+
+  test("should not map MO private data if existing MO does not have ch_reference", async () => {
+    if (appData.update && appData.update.review_managing_officers_individual) {
+      appData.update.review_managing_officers_individual[0].ch_reference = undefined;
+    }
+
+    mockIsActiveFeature.mockReturnValue(false);
+    mockGetManagingOfficersPrivateData.mockResolvedValue(MOCK_MANAGING_OFFICERS_PRIVATE_DATA);
+
+    await fetchManagingOfficersPrivateData(appData, req);
+    expect(mockGetManagingOfficersPrivateData).toHaveBeenCalled();
+    expect(appData.update?.review_managing_officers_individual?.[0].usual_residential_address).toBeFalsy();
```

Comment:
Does `toBeFalsy()` not expect the address to be boolean?

Reply:
everything is truthy in JavaScript :-)
maybe test for undefined instead?